packages/bsky/src/cache/read-through.ts

```diff
@@ -0,0 +1,131 @@
+import { CacheItem, RedisCache } from './redis'
+import { cacheLogger as log } from '../logger'
+
+export type CacheOptions<T> = {
+  staleTTL: number
+  maxTTL: number
+  namespace?: string
+  fetchMethod: (key: string) => Promise<T | null>
+  fetchManyMethod?: (keys: string[]) => Promise<Record<string, T | null>>
+}
+
+export class ReadThroughCache<T> {
+  constructor(public redisCache: RedisCache, public opts: CacheOptions<T>) {}
```

Comment:
Not critical at all, but it might be interesting to keep a couple counters on here for tracking the hit rate.

Reply:
only problem is that the read through cache gets reinstantiated every time a service is instantiated so those counters wouldn't mean much :/